src/dodal/devices/undulator.py

```diff
@@ -90,58 +124,125 @@ def __init__(
             else:
                 self.length = None
 
-        super().__init__(name)
+            if undulator_period is not None:
+                self.undulator_period, _ = soft_signal_r_and_setter(
+                    int, initial_value=undulator_period
+                )
+            else:
+                self.undulator_period = None
```

Comment:
Could: It might be nice to have a helper function like:
```python
def _make_signal_if_not_none(initial_value, type):
    if initial_value is None:
        return None
    signal, _ = soft_signal_r_and_setter(
        type, initial_value=initial_value
    )
    return signal
```
for this and poles/length